### PR TITLE
Add prefix / suffix interface

### PR DESCRIPTION
**Pull request by wsxiaoys**

### CODE REVIEW

File: crates/tabby/src/serve/completions.rs [hunk 1]
```diff
@@ -11,12 +11,24 @@ mod languages;
...
+    prompt: String,
+
     /// https://code.visualstudio.com/docs/languages/identifiers
     #[schema(example = "python")]
     language: Option<String>,
 
-    #[schema(example = "def fib(n):")]
-    prompt: String,
+    /// When segments are set, the `prompt` is ignored during the inference.
+    segments: Option<Segments>,
```
Comment (icycodes) [inline crates/tabby/src/serve/completions.rs:22]: Filed `segment` is optional but `prompt` is not here. So if we want FIM, it is also required to have a `prompt` field with empty value in request.   
I think using a non-optional `prompt` (prefix) and an optional `suffix` field should be better.

Comment (wsxiaoys) [reply crates/tabby/src/serve/completions.rs:22]: as said in comment，when segments present，prompt field is ignored.

Comment (icycodes) [reply crates/tabby/src/serve/completions.rs:22]: LGTM

**Comment by wsxiaoys** [comment]
close in favor of #183

**Pull request closed without merging**